### PR TITLE
ZCS-13901 method for unloading extension

### PR DESCRIPTION
**Pull request by RakeshAMore**
Added method for unloading individual extensions.

### CODE REVIEW

**Review by dasiyogesh — APPROVED** [review]

**Review by vishnukantjangid — APPROVED** [review]

**Review by rcyarrapothu — APPROVED** [review]

File: store/src/java/com/zimbra/cs/extension/ExtensionUtil.java [hunk 1]
```diff
@@ -236,6 +236,18 @@ public static synchronized void destroyAll() {
         sInitializedExtensions.clear();
     }
 
+    public static synchronized void destroy(String extName) {
+        ZimbraExtension ext = getExtension(extName);
+        try {
+            RedoableOp.deregisterClassLoader(ext.getClass().getClassLoader());
+            ext.destroy();
```
Comment (desouzas) [inline store/src/java/com/zimbra/cs/extension/ExtensionUtil.java:243]: If we're going to make use of this then the following method needs to be updated to keep track of which extension registered the specified path (no change to signature):
https://github.com/Zimbra/zm-mailbox/blob/develop/store/src/java/com/zimbra/cs/extension/ExtensionDispatcherServlet.java#L59

Then the following method needs to be updated to make sure all paths registered by the extension via the above register method are removed (no change to signature):
https://github.com/Zimbra/zm-mailbox/blob/develop/store/src/java/com/zimbra/cs/extension/ExtensionDispatcherServlet.java#L130

This needs to happen to make sure extensions that register multiple paths can unregister them all.

File: store/src/java/com/zimbra/cs/extension/ExtensionUtil.java [hunk 2]
```diff
@@ -236,6 +236,18 @@ public static synchronized void destroyAll() {
         sInitializedExtensions.clear();
     }
 
+    public static synchronized void destroy(String extName) {
+        ZimbraExtension ext = getExtension(extName);
```
Comment (desouzas) [inline store/src/java/com/zimbra/cs/extension/ExtensionUtil.java:240]: What happens if this is null in the next line? (npe)

File: store/src/java/com/zimbra/cs/extension/ExtensionUtil.java (hunk 1 above)
Comment (RakeshAMore) [reply store/src/java/com/zimbra/cs/extension/ExtensionUtil.java:243]: Ok let's connect to discuss this.

File: store/src/java/com/zimbra/cs/extension/ExtensionUtil.java (hunk 2 above)
Comment (RakeshAMore) [reply store/src/java/com/zimbra/cs/extension/ExtensionUtil.java:240]: Yes.

File: store/src/java/com/zimbra/cs/extension/ExtensionUtil.java (hunk 1 above)
Comment (RakeshAMore) [reply store/src/java/com/zimbra/cs/extension/ExtensionUtil.java:243]: Created a separate story for handling this.

**Pull request merged**